neurodamus/gj_user_corrections.py

```diff
@@ -0,0 +1,172 @@
+        Mechanisims = stochastic_mechs
+    if remove_channels == 'only_non_stoch':
+        Mechanisims = non_stochastic_mechs
+    if remove_channels:
+        logging.info("Removing channels type = " + remove_channels)
+        _perform_remove_channels(node_manager, Mechanisims)
+
+    if 'special_tag' in settings:
+        gjc = 0.1
+        logging.info("****\n**** special_tag ****\n****")
+
+    # load g_pas
+    if filename := settings.get('load_g_pas_file'):
+        processed_cells = _update_gpas(node_manager, filename, gjc,
+                                       settings.get("correction_iteration_load", -1))
+        all_ranks_total = int(MPI.allreduce(processed_cells, MPI.SUM))
+        logging.info(f"Update g_pas to fit {gjc} - file {filename} for {all_ranks_total} cells")
+
+    # load current clamps
+    holding_ic_per_gid = {}
+    if filename := settings.get('manual_MEComboInfo_file'):
+        # Oren's note: If I manually injecting different holding current for each cell,
+        # I will inject the current - the holding the emMEComboInfoFile
+        if settings.get('procedure_type') == 'find_holding_current':
+            raise Exception("not make any sense")
+        holding_ic_per_gid = _load_holding_ic(node_manager, filename, gjc=gjc)
+        all_ranks_total = int(MPI.allreduce(len(holding_ic_per_gid), MPI.SUM))
+        logging.info(f"Load holding_ic from manual_MEComboInfoFile {filename} "
+                     f"for {all_ranks_total} cells")
+
+    SEClamp_current_per_gid = {}
+    if settings.get('procedure_type') == 'find_holding_current' \
+            and isinstance(settings.get('vc_amp'), str):
+        logging.info("Find_holding_current - voltage file - {settings['vc_amp']}")
+        if not settings.get('disable_holding'):
+            logging.warning("Doing V_clamp and not disable holding!")
+
+        SEClamp_current_per_gid = _find_holding_current(node_manager, settings.get('vc_amp'))
+        _save_seclamps(SEClamp_current_per_gid, output_dir=SimConfig.output_root)
+
+    return holding_ic_per_gid, SEClamp_current_per_gid
+
+
+def _update_conductance(gjc, gj_manager):
+    n_conn = 0
+    for conn in gj_manager.all_connections():
+        conn.update_conductance(gjc)
+        n_conn += 1
+    return n_conn
+
+
+def _determanisitc_stoch(node_manager):
+    for cell in node_manager.cells:
+        for sec in cell._cellref.all:
+            if 'StochKv3' in dir(sec(.5)): sec.deterministic_StochKv3 = 1
+            if 'StochKv2' in dir(sec(.5)): sec.deterministic_StochKv2 = 1
+            if 'StochKv1' in dir(sec(.5)): sec.deterministic_StochKv1 = 1
+
+
+def _perform_remove_channels(node_manager, Mechanisms: list):
+    for cell in node_manager.cells:
+        for sec in cell._cellref.all:
+            for mec in Mechanisms:
+                if mec in dir(sec(.5)): sec.uninsert(mec)
+
+
+def _update_gpas(node_manager, filename, gjc, correction_iteration_load):
+    import h5py
```

Comment:
nitpick: maybe we would better import h5py at module level. It is not really optional, right?

Reply:
Not only here, we also import `h5py` at function level in other places in neurodamus because it triggers the numpy warnings. We have a filter to display the warning message once https://github.com/BlueBrain/neurodamus/blob/main/neurodamus/commands.py#L213 but it can only filter when the warning appears at function call.  I remember before I tried to move the filter even earlier in `init.py` but I got some issues with MPI calling before init Neuron.